nixos/modules/profiles/macos-builder.nix

```diff
@@ -24,156 +24,171 @@ in
     }
   ];
 
-  # The builder is not intended to be used interactively
-  documentation.enable = false;
-
-  environment.etc = {
-    "ssh/ssh_host_ed25519_key" = {
-      mode = "0600";
+  options.services.macos-builder = {
+    enable = mkEnableOption ''
+      Run a Linux builder on macOS.
+    '';
 
-      source = ./keys/ssh_host_ed25519_key;
+    diskSize = mkOption {
+      type = types.ints.positive;
+      description = "Disk size of the Linux builder's VM (in megabytes).";
+      default = 20 * 1024;
     };
 
-    "ssh/ssh_host_ed25519_key.pub" = {
-      mode = "0644";
-
-      source = ./keys/ssh_host_ed25519_key.pub;
+    memorySize = mkOption {
+      type = types.ints.positive;
+      description = "Memory size of the Linux builder's VM (in megabytes).";
+      default = 3 * 1024;
     };
   };
 
-  # DNS fails for QEMU user networking (SLiRP) on macOS.  See:
-  #
-  # https://github.com/utmapp/UTM/issues/2353
-  #
-  # This works around that by using a public DNS server other than the DNS
-  # server that QEMU provides (normally 10.0.2.3)
-  networking.nameservers = [ "8.8.8.8" ];
-
-  nix.settings = {
-    auto-optimise-store = true;
+  config = mkIf cfg.enable {
```

Comment:
This does not enable a system service or something resembling that. So if we need a flag for this at all, it should not be `services.macos-builder.enable`.

Instead, this module is a profile, which is just a set of unconditional definitions.

Why did you make the configuration conditional?